py/htm/examples/mnist.py

```diff
@@ -77,18 +77,30 @@ def load_images(file_name):
 # These parameters can be improved using parameter optimization,
 # see py/htm/optimization/ae.py
 # For more explanation of relations between the parameters, see 
-# src/examples/mnist/MNIST_CPP.cpp 
+# src/examples/mnist/MNIST_CPP.cpp
+# Note: 
+# Some of these default parameters are commented out, 
+# that is to tell the parameter optimization we want to fix those values
+# (eg. for they are dependent variables (like boostStrength & dutyCyclePeriod), 
+# should be const (synPermConnected), 
+# or we don't want to optimize those (columnDimensions)). Overall this reduces
+# there hyper-parameter search space, which leads to faster search and possibly
+# better values obtained. 
+# Note: 
+# TODO: The experiment should do a proper split to train/eval/test datasets. 
+# Currently we are using the out of sample test data in the hyperparameter search, 
+# so we are essentially overfitting! 
```

Comment:
The MNIST dataset only comes with train & test data...  I think that this is a known issue, that AI's overfit on this dataset.

Reply:
> I think that this is a known issue, that AI's overfit on this dataset. 

I'd say it's the authors responsibility not to bias the procedure (ie crossvalidation)

Reply:
> I'm going to write a few things for this README: 

great :+1:  The code needed some explaining, took me a bit trial-error to get it run.